storagemarket/deal_execution.go

```diff
@@ -88,14 +114,32 @@ func (p *Provider) execDealUptoAddPiece(ctx context.Context, pub event.Emitter,
 	// Publish
 	if deal.Checkpoint <= dealcheckpoints.Published {
 		if err := p.publishDeal(ctx, pub, deal); err != nil {
-			return fmt.Errorf("failed to publish deal: %w", err)
+			return &dealMakingError{
+				err: fmt.Errorf("failed to publish deal: %w", err),
+			}
 		}
 	}
+	// Now that the deal has been published, we no longer need to have funds
+	// tagged as being for this deal (the publish message moves collateral
+	// from the storage market actor escrow balance to the locked balance)
+	presp := make(chan struct{}, 1)
+	select {
+	case p.publishedDealChan <- publishDealReq{deal: deal, done: presp}:
+	case <-ctx.Done():
+		return &dealMakingError{recoverable: true, err: ctx.Err()}
```

Comment:
I don't think the context cancelled error is going to be very clear to the user in the UI, maybe for context cancelled we can have a message like "the deal was paused in the Publishing state because Boost was shut down"

Reply:
done.